<a href="https://colab.research.google.com/github/cloealberto/GapMinderAnalisePandas/blob/main/GapminderPandasEDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [18]:
# importando a biblioteca pandas
import pandas as pd

In [19]:
df = pd.read_csv('/content/Gapminder.csv', sep = ';',error_bad_lines=False) # error_bad_lines=False para ignorar as linhas com erros e ler o csv.

<ipython-input-19-9128a21c4e32>:1: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version. Use on_bad_lines in the future.


  df = pd.read_csv('/content/Gapminder.csv', sep = ';',error_bad_lines=False) # error_bad_lines=False para ignorar as linhas com erros e ler o csv.


In [20]:
# Visualizando as 5 primeiras linhas do dataframe
df.head()

country continent  year  lifeExp       pop   gdpPercap
0  Afghanistan      Asia  1952   28.801   8425333  779.445314
1  Afghanistan      Asia  1957   30.332   9240934  820.853030
2  Afghanistan      Asia  1962   31.997  10267083  853.100710
3  Afghanistan      Asia  1967   34.020  11537966  836.197138
4  Afghanistan      Asia  1972   36.088  13079460  739.981106

In [21]:
# Renomeando os nomes das colunas para fica mais claro.
df = df.rename(columns = {'country':'País', 'continent':'Continente', 'year':'Ano','lifeExp':'Expectativa de Vida', 'pop':'População','gdpPercap':'PIB Per Capita'})

In [27]:
#Verificando a estrutura de linhas e colunas do dataset
df_estrutura = df.shape
print("Esse dataset tem {} linhas e {} colunas.".format(df_estrutura[0], df_estrutura[1]))

Esse dataset tem 3312 linhas e 6 colunas


In [24]:
# Visualizando a alteração dos nomes das colunas e as 10 primeiras linhas do dataset
df.head(10)

País Continente   Ano  Expectativa de Vida  População  \
0  Afghanistan       Asia  1952               28.801    8425333   
1  Afghanistan       Asia  1957               30.332    9240934   
2  Afghanistan       Asia  1962               31.997   10267083   
3  Afghanistan       Asia  1967               34.020   11537966   
4  Afghanistan       Asia  1972               36.088   13079460   
5  Afghanistan       Asia  1977               38.438   14880372   
6  Afghanistan       Asia  1982               39.854   12881816   
7  Afghanistan       Asia  1987               40.822   13867957   
8  Afghanistan       Asia  1992               41.674   16317921   
9  Afghanistan       Asia  1997               41.763   22227415   

   PIB Per Capita  
0      779.445314  
1      820.853030  
2      853.100710  
3      836.197138  
4      739.981106  
5      786.113360  
6      978.011439  
7      852.395945  
8      649.341395  
9      635.341351

In [37]:
# Visualizando as 10 últimas linhas do dataset
df.tail(10)

País Continente   Ano  Expectativa de Vida  População  \
3302  Zimbabwe     Africa  1962               52.358    4277736   
3303  Zimbabwe     Africa  1967               53.995    4995432   
3304  Zimbabwe     Africa  1972               55.635    5861135   
3305  Zimbabwe     Africa  1977               57.674    6642107   
3306  Zimbabwe     Africa  1982               60.363    7636524   
3307  Zimbabwe     Africa  1987               62.351    9216418   
3308  Zimbabwe     Africa  1992               60.377   10704340   
3309  Zimbabwe     Africa  1997               46.809   11404948   
3310  Zimbabwe     Africa  2002               39.989   11926563   
3311  Zimbabwe     Africa  2007               43.487   12311143   

      PIB Per Capita  
3302      527.272182  
3303      569.795071  
3304      799.362176  
3305      685.587682  
3306      788.855041  
3307      706.157306  
3308      693.420786  
3309      792.449960  
3310      672.038623  
3311      469.709298

In [35]:
#Printando o nome das colunas
df.columns

Index(['País', 'Continente', 'Ano', 'Expectativa de Vida', 'População',
       'PIB Per Capita'],
      dtype='object')

In [36]:
# Tipos de dados das colunas
df.dtypes

País                    object
Continente              object
Ano                      int64
Expectativa de Vida    float64
População                int64
PIB Per Capita         float64
dtype: object

In [39]:
# Informações estatísticas do conjunto de dados
df.describe()

Ano  Expectativa de Vida     População  PIB Per Capita
count  3312.000000          3312.000000  3.312000e+03     3312.000000
mean   1980.301630            65.246871  3.161489e+07    11317.115805
std      16.927294            11.768412  1.041193e+08    11369.142739
min    1950.000000            23.599000  5.941200e+04      241.165876
25%    1967.000000            58.337500  2.678572e+06     2514.625266
50%    1982.000000            69.610000  7.557218e+06     7838.505123
75%    1996.000000            73.657750  1.958522e+07    17357.879170
max    2007.000000            82.670000  1.318683e+09   113523.132900

**Algumas observações nesse describe:**
São 3312 registros.
A média de expectativa de vida é de 65,2 anos.
O PIB Per Capita médio é 11.317,12.


In [40]:
# Verficando os continentes do conjunto de dados
df['Continente'].unique()

array(['Asia', 'Europe', 'Africa', 'Americas', nan, 'FSU', 'Oceania'],
      dtype=object)

In [42]:
# Usando loc para criar um dataset a parte para Oceania
Oceania = df.loc[df['Continente'] == 'Oceania']
Oceania.head(10)

País Continente   Ano  Expectativa de Vida  População  PIB Per Capita
975  Fiji    Oceania  1962               56.494     421869     2039.393277
976  Fiji    Oceania  1967               58.607     485000     2170.073285
977  Fiji    Oceania  1972               60.642     544000     2797.759484
978  Fiji    Oceania  1977               62.669     599339     3182.569600
979  Fiji    Oceania  1982               64.683     658906     3209.257468
980  Fiji    Oceania  1987               66.626     719574     2974.850227
981  Fiji    Oceania  1992               66.575     750242     3410.286040
982  Fiji    Oceania  1997               66.836     797890     3643.629208
983  Fiji    Oceania  2002               67.832     856346     4004.481972
984  Fiji    Oceania  2007               68.774     918675     4118.158294

In [43]:
# Verficando os países dentro do dataset Oceania
Oceania['País'].unique()

array(['Fiji', 'New Zealand', 'Solomon Islands'], dtype=object)

In [61]:
# Agrupando o total de país no continente Oceania
Total_paises = Oceania.groupby('Continente')['País'].nunique()
print('O total de países da Oceania disponíveis nesse dataset é {}.'.format(Total_paises[0]))

O total de países da Oceania disponíveis nesse dataset é 3.


In [63]:
# Total de países por continente no geral
df.groupby('Continente')['País'].nunique()

Continente
Africa      51
Americas    25
Asia        41
Europe      35
FSU          6
Oceania      3
Name: País, dtype: int64

In [69]:
# Qual a expectativa de vida, na média, ao longo dos anos?
round(df.groupby('Ano')['Expectativa de Vida'].mean(),1)

Ano
1950    62.0
1951    65.9
1952    49.2
1953    66.7
1954    67.5
1955    67.8
1956    68.0
1957    51.6
1958    68.8
1959    68.2
1960    68.5
1961    68.9
1962    54.0
1963    69.6
1964    70.1
1965    70.3
1966    70.4
1967    56.3
1968    70.7
1969    70.7
1970    71.0
1971    71.1
1972    58.5
1973    71.5
1974    71.8
1975    71.9
1976    72.2
1977    60.4
1978    72.7
1979    73.0
1980    73.1
1981    73.3
1982    62.4
1983    73.8
1984    74.1
1985    74.1
1986    74.5
1987    64.0
1988    74.8
1989    74.9
1990    74.3
1991    74.4
1992    65.0
1993    74.3
1994    74.5
1995    74.6
1996    75.0
1997    65.9
1998    75.6
1999    75.7
2000    76.0
2001    76.3
2002    66.8
2003    76.6
2004    76.9
2005    76.7
2006    77.9
2007    67.9
Name: Expectativa de Vida, dtype: float64